### PR TITLE
Use different icon for some file types when brave is set as a default application

### PR DESCRIPTION
**Pull request by simonhong**
fix https://github.com/brave/brave-browser/issues/12761

When brave browser is set as a default application,
different icon will be used for that file type.
With this PR, `.pdf` and `.svg` files will have different icon.

image
image

So far, we set `BraveBHTML` (beta) prog id for supported file types in registry like below.
image
and that `BraveBHTML` prog id entry has some attributes such as default icon or commands.
That default icon is used for th file type when browser is set as its default application.
default icon of `BraveBHTML`  reg entry is brave browser's default icon.
So, all file types uses same icon.
image

This PR introduces new `BraveFile` prog entry and it uses different icon.
With this PR, new `BraveFile` id is set for `pdf` like below.
image
and new `BraveFile` entry is newly added.
image

Resolves 

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Download pdf or svg file
2. Set Brave as their default application
3. Check these files' icon is set to brave's file icon(not brave default icon)
4. Uninstall brave and check these files' icon is set properly to other application.

### CODE REVIEW

File: chromium_src/chrome/installer/setup/uninstall.cc [hunk 1]
```diff
@@ -0,0 +1,68 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/installer/util/brave_shell_util.h"
```
Comment (mariospr) [inline chromium_src/chrome/installer/setup/uninstall.cc:6]: Don't you need to add another dependency on `//brave/installer/util` from the `//chrome/installer/setup:setup` GN target for this?

Comment (simonhong) [reply chromium_src/chrome/installer/setup/uninstall.cc:6]: Oops, I missed that. Added (https://github.com/brave/brave-core/pull/9056/files#diff-b59cb8001bdafc216af8342ad837f33188b3673c1bbd13ff04794e3b3afc4988R9)

File: patches/chrome-installer-setup-BUILD.gn.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
+diff --git a/chrome/installer/setup/BUILD.gn b/chrome/installer/setup/BUILD.gn
+index db3796b6b6ef7c7c60cd6d75e93259c1a5edc364..60e23ef9d63ba15a3c54bdb9674615ee1fa20dbf 100644
+--- a/chrome/installer/setup/BUILD.gn
++++ b/chrome/installer/setup/BUILD.gn
+@@ -49,6 +49,8 @@ if (is_win) {
+     ]
+ 
+     libs = [ "netapi32.lib" ]
++    import("//brave/chromium_src/chrome/installer/setup/sources.gni") deps += brave_chromium_src_chrome_installer_setup_deps
++
```
Comment (mariospr) [inline patches/chrome-installer-setup-BUILD.gn.patch:10]: Nit: unnecessary extra blank line :innocent:

**Review by mariospr — APPROVED** [review]
I'm not very familiar with Windows-specific bits, but the PR lgtm from the pov of patching (even though it's unfortunate it introduces 3 new patches, but I can't see an alternative)

File: patches/chrome-installer-setup-BUILD.gn.patch (hunk 2 above)
Comment (simonhong) [reply patches/chrome-installer-setup-BUILD.gn.patch]: Deleted. thanks!

**Comment by simonhong** [comment]
Kindly ping to @brave/chromium-src-reviewers @brave/deps-reviewers :)

**Review by iefremov — APPROVED** [review]
DEPS lgtm

**Comment by simonhong** [comment]
Merged because only unrelated `BatAdsCreateConfirmationUrlRequestBuilderTest` were failed.

**Pull request merged**